apps/api/src/modules/job/application/offer/offer.repository.ts

```diff
@@ -0,0 +1,40 @@
+import { ApplicationStatus } from '@modela/database'
+import { Injectable } from '@nestjs/common'
+import { PrismaService } from 'src/database/prisma.service'
+
+import { ApplicationRepository } from '../application.repository'
+
+@Injectable()
+export class OfferRepository {
+  constructor(
+    private prisma: PrismaService,
+    private appRepo: ApplicationRepository,
+  ) {}
+
+  async updateApplicationStatus(
+    applicationId: number,
+    status: ApplicationStatus,
+  ) {
+    await this.prisma.application.update({
+      where: {
+        applicationId: applicationId,
+      },
+      data: {
+        status: status,
+      },
+    })
+    return
+  }
+
+  async getJobById(jobId: number) {
+    return await this.prisma.job.findFirst({
+      where: {
+        jobId: jobId,
+      },
+    })
+  }
```

Comment:
use getBaseJobById in jobRepository
and actually you should use findUnique not findFirst